OPHD/UI/IconGrid.cpp

```diff
@@ -371,11 +371,61 @@ void IconGrid::update()
 		// Name Tooltip
 		if (mShowTooltip)
 		{
-			const auto& highlightedName = mIconItemList[mHighlightIndex].name;
-			const auto tooltipRect = NAS2D::Rectangle<int>{{position.x, position.y - 15}, {mFont.width(highlightedName) + 4, mFont.height()}};
-			renderer.drawBoxFilled(tooltipRect, NAS2D::Color{245, 245, 245});
-			renderer.drawBox(tooltipRect, NAS2D::Color{175, 175, 175});
-			renderer.drawText(mFont, highlightedName, position + NAS2D::Vector{2, -15}, NAS2D::Color::Black);
+			std::vector<std::string> toolTipStrings;
+			toolTipStrings.push_back(mIconItemList[mHighlightIndex].name);
+			/* HEADS UP Feels like there is a better way. Unless more robots are to be added it works.
+			* Without this name check accessing the structure resource and population requirements causes an error because the robots are not structures*/
+			if (toolTipStrings[0] != "Robodigger" && toolTipStrings[0] != "Robodozer" && toolTipStrings[0] != "Robominer" && this->mIconSheet.name() != "ui/factory.png")
+			{
+				const auto& tempStructure = Structure::Structure(Structure::StructureClass::Undefined, (StructureID)mIconItemList[mHighlightIndex].meta);
+				const auto& popNeeds = tempStructure.populationRequirements();
+				const auto& cost = StructureCatalogue::costToBuild((StructureID)mIconItemList[mHighlightIndex].meta);
```

Comment:
Similar to the last comment, this might be easier to lookup when setting the grid data, rather than during display.